compiler/rustc_mir_dataflow/src/value_analysis.rs

```diff
@@ -718,6 +728,25 @@ impl Map {
         });
     }
 
+    /// Precompute the list of values inside `root` and store it inside
+    /// as a slice within `inner_values_buffer`.
+    fn cache_preorder_invoke(&mut self, root: PlaceIndex) {
+        let start = self.inner_values_buffer.len();
+        if let Some(vi) = self.places[root].value_index {
+            self.inner_values_buffer.push(vi);
+        }
+
+        // We manually iterate instead of using `children` as we need to mutate `self`.
+        let mut next_child = self.places[root].first_child;
+        while let Some(child) = next_child {
+            self.cache_preorder_invoke(child);
```

Comment:
how deep can this recursion get? We can have some very long projections from user code. A recursion limit could either cause a miscompilation or a hard error where there shouldn't be one, so maybe use `ensure_sufficient_stack`?